### PR TITLE
vendor: github.com/containerd/containerd v1.6.3

### PR DESCRIPTION
**Pull request by thaJeztah**
full diff: https://github.com/containerd/containerd/compare/v1.6.2...v1.6.3

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: vendor/github.com/containerd/containerd/pkg/kmutex/kmutex.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+	locks map[string]*klock
+}
+
+type klock struct {
+	*semaphore.Weighted
+	ref int
+}
+
+func (km *keyMutex) Lock(ctx context.Context, key string) error {
```
Comment (thaJeztah) [inline vendor/github.com/containerd/containerd/pkg/kmutex/kmutex.go:55]: This package looks awfully close to https://github.com/moby/locker - baring the `context` (which it doesn't have); wondering if we should swap one for the other @cpuguy83 ?

**Review by AkihiroSuda — APPROVED** [review]

**Comment by thaJeztah** [comment]
I'll rebase this one

**Comment by Sagor2214** [comment]
I'll rebase this one

**Comment by thaJeztah** [comment]
^^ blocked and reported for abuse

**Review by tianon — APPROVED** [review]

**Pull request merged**